### PR TITLE
Add additional filtering capabilities to the scheduler api for queues and tasks endpoints

### PR DESCRIPTION
**Pull request by jpbruinsslot**
Adds support for more complex queries to the queues and task list endpoints of the scheduler. If usage of json payload to filter is undesirable we can move the payload to the GET parameters when needed.

Examples:

* Filter on nested fields, using "`__`" as separator:

```javascript
POST /tasks

{
    "filters": [
        {
            "column": "p_item",
            "field": "data__boefje__id",
            "operator": "eq",
            "value": "dns-records"
        }
    ]
}
````

* Or on top level fields:

```javascript
POST /tasks

{
    "filters": [
        {
            "column": "status",
            "operator": "eq",
            "value": "completed"
        }
    ]
}
````

* Support for `and`, `or` and `not` filter operators:

```javascript
POST /tasks

{
    "filters": {
        "or": [
            {
                "column": "p_item",
                "field": "data__boefje__id",
                "operator": "eq",
                "value": "dns-records"
            },
            {
                "column": "p_item",
                "field": "data__normalizer__id",
                "operator": "eq",
                "value": "kat_manual_csv",
            }
        ]
}
```

* Support for the following comparison operators:

```
==
eq
!=
ne
is
is_not
is_null
is_not_null
>
gt
<
lt
>=
gte
<=
lte
like
not_like
ilike
not_ilike
in
not_in
contains
any
match
starts_with
@>
<@
@?
@@
```

```javascript
POST /tasks

{
    "filters": [
        {
            "column": "p_item",
            "field": "data__input_ooi",
            "operator": "like",
            "value": "%internet%"
        },
        {
            "column": "p_item",
            "field": "data__boefje__id",
            "operator": "eq",
            "value": "dns-zone"
        }
    ]
}

```

### CODE REVIEW

**Comment by jpbruinsslot** [comment]
Note that I would like to implement backwards compatibility for how the prior filtering worked, such that rocky will still be able use that endpoint and that it eventually can migrate to the new filtering.

**Review by Donnype — COMMENTED** [review]
We could also just POST raw SQL right? Jk 😄 

I think given the current setup this gives a lot more options in the API. I do wonder if we need everything to be this generic. E.g. do we really want to do complex nested field filtering vs. fetching the related object and querying the target object on the foreign key id?  (So for example `task = get(task, name="a"); get(item, task_id=task.id)` versus `get(item, task__name=="a")`.) Because the logic regarding SqlAlchemy is starting to diverge from how you would typically use it to provide the generic filters through the API.

File: mula/scheduler/storage/filters/functions.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+from sqlalchemy.orm.query import Query
+from sqlalchemy.sql.elements import BinaryExpression
+
+from .casting import cast_expression
+from .comparison import Comparator
+from .filters import FilterRequest
+from .operators import FILTER_OPERATORS
+
+
+def apply_filter(entity, query: Query, filter_request: FilterRequest):
```
[the remark this answers is not in the document]
Comment (jpbruinsslot) [reply mula/scheduler/storage/filters/functions.py:10]: My consideration here is that I apply the filter request to the query object that is passed. Does your suggestion mean creating the `apply_to` method with the query as an argument?

**Comment by jpbruinsslot** [comment]
> We could also just POST raw SQL right? Jk 😄
> 
> I think given the current setup this gives a lot more options in the API. I do wonder if we need everything to be this generic. E.g. do we really want to do complex nested field filtering vs. fetching the related object and querying the target object on the foreign key id? (So for example `task = get(task, name="a"); get(item, task_id=task.id)` versus `get(item, task__name=="a")`.) Because the logic regarding SqlAlchemy is starting to diverge from how you would typically use it to provide the generic filters through the API.

Thanks! Would you mind discussing this a bit further in a call? @Donnype

File: mula/scheduler/storage/filters/functions.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Donnype) [reply mula/scheduler/storage/filters/functions.py:10]: Yes so only dropping the `filter_request` in favor of a `self`, but it's a matter of taste as well :)

**Comment by Darwinkel** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [ ] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
- Overall, filtering works very well!
- No apparent regressions

### What doesn't work:
- Some (developer) docs and/or examples regarding the filtering semantics (e.g. what happens when doing a `gt` or `lte` search on a string) and a showcase of the search features would be nice.
- I identified and reported a bunch of unhandled edge cases and semantic details @jpbruinsslot

File: mula/scheduler/server/server.py [hunk 2]
```diff
@@ -381,6 +381,8 @@ def list_tasks(
 
             f_req.filters.update(f_plugin)  # type: ignore
 
+        self.logger.info(f"FILTER_REQUEST: {f_req}")
```
Comment (jpbruinsslot) [inline mula/scheduler/server/server.py:384]: Remove

**Pull request merged**